qiskit/optimization/algorithms/minimum_eigen_optimizer.py

```diff
@@ -131,6 +131,26 @@ def get_compatibility_msg(self, problem: QuadraticProgram) -> str:
         """
         return QuadraticProgramToQubo.get_compatibility_msg(problem)
 
+    @property
+    def min_eigen_solver(self) -> Optional[MinimumEigensolver]:
+        """Returns the minimum eigensolver."""
+        return self._min_eigen_solver
+
+    @min_eigen_solver.setter
+    def min_eigen_solver(self, min_eigen_solver: MinimumEigensolver) -> None:
+        """Sets the minimum eigensolver."""
+        self._min_eigen_solver = min_eigen_solver
+
+    @property
+    def penalty(self) -> Optional[float]:
+        """Returns the penalty value."""
+        return self._penalty
+
+    @penalty.setter
+    def penalty(self, penalty: float) -> None:
```

Comment:
```suggestion
    def penalty(self, penalty: Optional[float]) -> None:
```
As the constructor allows passing in None, which then uses some default computed penalty, a user might want to revert back to None having previously set some specific value, so allowing None here on the setter i.e. Optional, seems like the ting to do.